platform/src/main/java/org/stellar/anchor/platform/utils/RequestLoggerFilter.java

```diff
@@ -0,0 +1,98 @@
+package org.stellar.anchor.platform.utils;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.Objects;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+import org.stellar.anchor.util.Log;
+
+/**
+ * Log each request and response. The request body is never logged and the response body is only
+ * logged when it's an error
+ *
+ * @see <a href="https://stackoverflow.com/a/42023374/875657">StackOverflow Answer</a>
+ */
+public class RequestLoggerFilter extends OncePerRequestFilter {
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws ServletException, IOException {
+
+    long startTime = System.currentTimeMillis();
+    StringBuffer logMessage =
+        new StringBuffer("\n\tREST Request:\n")
+            .append("\t\t[METHOD: " + request.getMethod() + "]\n")
+            .append("\t\t[PATH: " + request.getRequestURI());
+
+    if (request.getQueryString() != null) {
+      logMessage.append("?").append(request.getQueryString());
+    }
+    logMessage.append("]\n");
+    if (request.getAuthType() != null) {
+      logMessage.append("\t\t[AUTH_TYPE: " + request.getAuthType() + "]\n");
+    }
+    if (request.getUserPrincipal() != null) {
+      logMessage.append("\t\t[PRINCIPAL_NAME: " + request.getUserPrincipal().getName() + "]\n");
+    }
+    logMessage.append("\t\t[CLIENT IP: " + request.getRemoteAddr() + "]\n");
```

Comment:
will this be the actual client's IP or just the IP of the reverse proxy that sits in front of Anchor Platform?

Reply:
Great catch! After some research, I've found out that it could be displaying the address of a proxy server instead of the actual client IP. I'll need to fill that manually.

Thanks for spotting it!

Reply:
Done in https://github.com/stellar/java-stellar-anchor-sdk/pull/519/commits/12a895820de1130e0726b0a0a80f5950126dcb3a.